src/Middleware/BasicNetworkResolver.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+
+namespace Yiisoft\Yii\Web\Middleware;
+
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Basic network resolver
+ *
+ * It can be used in the following cases:
+ * - not required IP resolve to access the user's IP
+ * - user's IP is already resolved (eg `ngx_http_realip_module` or similar)
+ *
+ * @package Yiisoft\Yii\Web\Middleware
+ */
+class BasicNetworkResolver implements MiddlewareInterface
+{
+    private const DEFAULT_PROTOCOL_AND_ACCEPTABLE_VALUES = [
+        'http' => ['http'],
+        'https' => ['https', 'on'],
+    ];
+
+    private $protocolHeaders = [];
+
+    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
+    {
+        $newScheme = null;
+        foreach ($this->protocolHeaders as $header => $data) {
+            if (!$request->hasHeader($header)) {
+                continue;
+            }
+            $headerValues = $request->getHeader($header);
+            if (is_callable($data)) {
+                $newScheme = call_user_func($data, $headerValues, $header, $request);
+                if ($newScheme === null) {
+                    continue;
+                } elseif (!is_string($newScheme)) {
```

Comment:
the `elseif` isn't needed here, can be a simple if

Reply:
modified